network/notifee.go

```diff
@@ -84,6 +84,10 @@ func (s *NotifeeService) Stop() {
 	s.lp2pEventSub.Close()
 }
 
+func (s *NotifeeService) Reachability() lp2pnetwork.Reachability {
+	return s.reachability
```

Comment:
consider adding an RLock here to avoid the deadlock issue.

Reply:
No need, This Will call only be the external API and only change once.

Reply:
The Readability is set only once, and the external API calls it. It is not necessary to add a Lock here. We have a test failure. Let's fix that.

Reply:
So, why there is a deadlock issue?